File: CS/MyRestService/WebApiDataSourceLoadOptions.cs
using DevExtreme.AspNet.Data;
using DevExtreme.AspNet.Data.Helpers;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Http.ModelBinding;
using System.Web.Http.Controllers;

namespace MyRestService {


    [ModelBinder(typeof(WebApiDataSourceLoadOptionsBinder))]
    public class WebApiDataSourceLoadOptions : DataSourceLoadOptionsBase {
    }


    public class WebApiDataSourceLoadOptionsBinder : IModelBinder {

        public bool BindModel(HttpActionContext actionContext, ModelBindingContext bindingContext) {
            var loadOptions = new WebApiDataSourceLoadOptions();

            DataSourceLoadOptionsParser.Parse(loadOptions, key => {
                var value = bindingContext.ValueProvider.GetValue(key);
                if(value != null)
                    return value.AttemptedValue;
                return null;
            });

            bindingContext.Model = loadOptions;
            return true;
        }

    }


}

File: VB/MyRestService/WebApiDataSourceLoadOptions.vb
Imports DevExtreme.AspNet.Data
Imports DevExtreme.AspNet.Data.Helpers
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Http.ModelBinding
Imports System.Web.Http.Controllers

Namespace MyRestService
    <ModelBinder(GetType(WebApiDataSourceLoadOptionsBinder))>
    Public Class WebApiDataSourceLoadOptions
        Inherits DataSourceLoadOptionsBase
    End Class


    Public Class WebApiDataSourceLoadOptionsBinder
        Implements IModelBinder

        Public Function BindModel(actionContext As HttpActionContext, bindingContext As ModelBindingContext) As Boolean Implements IModelBinder.BindModel
            Dim loadOptions = New WebApiDataSourceLoadOptions()
            DataSourceLoadOptionsParser.Parse(loadOptions, Function(key)
                                                               Dim value = bindingContext.ValueProvider.GetValue(key)
                                                               If value IsNot Nothing Then
                                                                   Return value.AttemptedValue
                                                               End If
                                                               Return Nothing
                                                           End Function)
            bindingContext.Model = loadOptions
            Return True
        End Function

    End Class
End Namespace

